src/vm/gchandleutilities.h

```diff
@@ -33,6 +33,10 @@ class GCHandleUtilities
 void ValidateObjectAndAppDomain(OBJECTREF objRef, ADIndex appDomainIndex);
 void ValidateHandleAssignment(OBJECTHANDLE handle, OBJECTREF objRef);
 
+// Check that the supplied object is valid to put in a pinned handle,
+// throwing an exception if not.
+void ValidatePinnedObject(OBJECTREF obj);
```

Comment:
Shouldn't this rather be in marshalnative.h since the implementation is in marshalnative.cpp; and it implements arbitrary interop limitation - nothing fundamental for the GC handle table?